test/e2e/storage/csi_mock/csi_volume_expansion.go

```diff
@@ -17,17 +17,57 @@ limitations under the License.
 package csi_mock
 
 import (
+	"context"
+	"fmt"
+	"time"
+
+	csipbv1 "github.com/container-storage-interface/spec/lib/go/csi"
 	"github.com/onsi/ginkgo/v2"
 	"github.com/onsi/gomega"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
 	"k8s.io/kubernetes/test/e2e/storage/testsuites"
 	"k8s.io/kubernetes/test/e2e/storage/utils"
 	admissionapi "k8s.io/pod-security-admission/api"
 )
 
+type expansionStatus int
+
+const (
+	expansionSuccess = iota
+	expansionFailed
+	expansionFailedOnController
+	expansionFailedOnNode
+)
+
+const (
+	resizePollInterval = 2 * time.Second
+)
+
+var (
+	maxControllerSizeLimit = resource.MustParse("10Gi")
+
+	maxNodeExpansionLimit = resource.MustParse("8Gi")
+)
+
+type recoveryTest struct {
+	name                    string
+	pvcRequestSize          string
+	allocatedResource       string
+	expectedExpansionStatus expansionStatus
+	resizeStatus            v1.PersistentVolumeClaimResizeStatus
```

Comment:
nit: this should be expectedResizeStatus (or expectedFirstResizeStatus to emphasize it's the first one?)